src/components/add-challenge/AddChallengeForm.tsx

```diff
@@ -0,0 +1,289 @@
+"use client"
+
+import { CreateChallengeRequest, Hint } from "@/types/challenges.types";
+import { Input, Text, Box, Textarea, NumberInputField, NumberInputStepper, NumberDecrementStepper, NumberInput, NumberIncrementStepper, Select, Button, Alert, AlertIcon, AlertTitle, AlertDescription, useToast } from "@chakra-ui/react";
+import { useMutation } from "@tanstack/react-query";
+import * as React from "react";
+import { useState } from "react";
+
+export default function AddChallengeForm(){
+    type CreateChallengeFormData = {
+        challenge_name: string;
+        points: number;
+        challenge_description: string;
+        flag: string;
+        is_flag_case_sensitive: boolean;
+        division: string;
+        challenge_category: string;
+        solution_explanation: string;
+        hints: Hint[];
+        // file_attachment: File | null;
+    };
```

Comment:
I don't wanna be annoying but can we also move this type to types folder to keep the component cleaner?

Reply:
yeah, i thought the same but i think it's fine since it's a type specific to that component and wouldn't be used anywhere else

Reply:
changed my mind...decided to move it